ui/api/api.ts

```diff
@@ -0,0 +1,26 @@
+import { getUser } from "@/utils/session";
+import { z } from "zod";
+
+export async function getHeaders(): Promise<Record<string, string>> {
+  const user = await getUser();
+  return {
+    Accept: "application/json",
+    Authorization: `Bearer ${user.accessToken}`,
+  };
+}
+
+export const BackendError = z.object({
+  meta: z.object({ type: z.string() }), // z.map(z.string(), z.string()),
```

Comment:
`meta` could probably be `z.any`. I didn't realize that was available when I added `BackendError` in another PR.

Reply:
Is there a unified error response from the BE?

Reply:
There are some responses like for topics where there could be errors embedded - i.e. either the list of partitions or an error. This is to support situations where the user can list topics but cannot describe one or more of them. We can perhaps adjust this either/or approach if it proves unwieldy or other just ugly. It's just something I made up to support returning _something_ when something else failed during a composite operation.

Then there is a general errors response for things like validation or Kafka errors that affect the entire request. Think 404 for fetching a single topic or bad request parameters. That response follows the JSON API errors response `{ "errors": [ /* list of errors */ ] }`. The error objects are the same in either case - they match the `BackendError` zod schema.